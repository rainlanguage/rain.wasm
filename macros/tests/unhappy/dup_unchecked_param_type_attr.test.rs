#[macro_use]
extern crate wasm_bindgen_utils_macros;

struct TestStruct;

#[wasm_export]
impl TestStruct {
    #[wasm_export]
    pub async fn test_method(
        #[wasm_export(unchecked_param_type = "FirstType", unchecked_param_type = "SecondType")]
        arg: wasm_bindgen::JsValue
    ) -> Result<String, Error> {
        Ok("test".to_string())
    }
}

fn main() {}