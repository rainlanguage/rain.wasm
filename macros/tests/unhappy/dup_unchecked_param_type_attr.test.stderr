error: duplicate `unchecked_param_type` attribute
  --> tests/unhappy/dup_unchecked_param_type_attr.test.rs:10:59
   |
10 |         #[wasm_export(unchecked_param_type = "FirstType", unchecked_param_type = "SecondType")]
   |                                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
